app/assets/javascripts/host_edit.js

```diff
@@ -295,10 +295,7 @@ function serializeForm() {
 
 //This function decides for a given input element is it set explicitly by the user.
 function isExplicit(index, element) {
-  if (element.nextSibling == undefined) return true;
-  if (element.nextSibling.children == undefined) return true;
-  if (element.nextSibling.children[0] == undefined) return true;
-  if (element.nextSibling.children[0].dataset == undefined) return true;
+  if (!element.classList.contains("has-explicit")) return true;
   return element.nextSibling.children[0].dataset.explicit == 'true';
```

Comment:
If the class is on the button (as opposed to `element`), I will have to write defensive code (like it was before), because the structure of serializable elements could vary. Basically anything except `select` elements won't have the same structure (neighbor `<span>` with `<button>` child).